File: demotdd/src/test/java/br/org/demotdd/calculadora/CalculadoraTest.java
package br.org.demotdd.calculadora;

import br.org.demotdd.calculadora.Calculadora;
import br.org.demotdd.calculadora.NaoPoderDividirPorZero;
import org.junit.Before;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class CalculadoraTest {

    private Calculadora calc;

    @Before
    public void setup(){
        calc = new Calculadora();
    }

    @Test
    public void deveSomarDoisNumeros(){
        int a = 20;
        int b = 15;
        int result = calc.somar(a, b);
        assertEquals(35,result);
      }

    @Test
    public void deveSubrairDoisNumeros(){
        int a = 15;
        int b = 15;
        int result = calc.subtrair(a, b);
        assertEquals(0,result);
    }

    @Test
    public void deveDividirDoisNumeros() throws NaoPoderDividirPorZero {
        int a = 15;
        int b = 15;
        int result = calc.dividir(a, b);
        assertEquals(1,result);
    }
    @Test(expected = NaoPoderDividirPorZero.class)
     public void deveLancharExcepion() throws NaoPoderDividirPorZero {
        int a = 15;
        int b = 0;
        int result = calc.dividir(a, b);
    }
}

File: tdd/src/test/kotlin/CalculadoraTest.kt
import org.junit.Assert.assertEquals
import org.junit.Test


class CalculadoraTest {

    @Test
    fun meuPrimeiroTeste() {
        val soma = 10 + 4
        assertEquals(14, soma)
    }

    @Test
    fun meuSegundoTeste() {
        //Arrange
        val calculadora = Calculadora()
        //ACT
        val soma = calculadora.somar(2, 3)
        //ASSERT
        assertEquals(5, soma)
    }

    @Test(expected = ParametroNegativoExcepion::class)
    fun sePrimeiroParametroForNegativoDeveLancarExcecao() {
        val calculadora = Calculadora()
        val soma = calculadora.somar(-2, 3)

    }


}